presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -399,19 +401,31 @@ private synchronized QueryResults getNextResultWithRetry(long token, UriInfo uri
         boolean hasNotRetried = queryManager.getQueryRetryCount(queryId) < 1;
 
         if (historyBasedOptimizationEnabled && hasNotRetried && retryConditionsMet(queryResults) && retryQueryWithHistoryBasedOptimizationEnabled(session)) {
+            WarningCode warningFromOriginalQuery = new WarningCode(700, "Your query is being retried");
```

Comment:
Don't we use constants for this WarningCode or something?

Reply:
Yeah you are right, I'll make it an enum